### PR TITLE
fix(ci): preserve canonical auto-heal candidates

### LINKED ISSUE #2059: fix(auto-heal): preserve canonical paths for ambiguous changed modules
## Problem

PDD PR #2058 auto-heal failed before healing because the changed-file candidate mapper emitted the short module name `cli` for a nested canonical file:

- Workflow: https://github.com/promptdriven/pdd/actions/runs/29380783150
- Cloud Build: `1fac527b-fe38-4e50-93b9-dbba649d95d6`
- Failure: `detect_drift failed: classification failed for cli/python: Ambiguous module cli ... pdd/core/cli.py ... pdd/cli.py`

The same log shows `commands/generate` and `core/cli` can resolve correctly when their canonical paths are retained. The failure is therefore in auto-heal changed-file candidate normalization, which also emitted other lossy aliases such as `commands_generate`, `core_dump`, and `core_errors`.

Closed issue #1677 established that PDD should resolve modules by canonical path instead of ambiguous short basename, but this auto-heal entrypoint still violates that invariant. This is a regression/sibling path, not evidence that #2058’s estimate fix is wrong.

## Acceptance criteria

- Inspect #1677 and related commits/PRs before designing the fix.
- Add a failing test for a PR changing both/nested CLI paths where short basenames collide.
- Auto-heal must preserve path-qualified canonical module identities from changed files/architecture.json.
- Reject genuinely ambiguous/unmapped candidates with an actionable diagnostic; do not guess.
- Cover nested command/core modules, prompt/code/test ownership mapping, and duplicate basenames.
- Run the exact auto-heal candidate/detect-drift boundary plus hosted auto-heal.

## Release impact

This is a required-check blocker for PR #2058. Land independently, integrate main into #2058, then rerun its exact-head hosted checks.

### PR DESCRIPTION
**Pull request by gltanaka**
## Summary

- resolve changed code, prompt, context, and flattened test candidates through canonical architecture ownership
- fall back only to checked-in canonical prompt/code pairs for legacy units missing architecture rows
- preserve disambiguating root identities such as `pdd/cli` alongside `core/cli`
- carry those path-qualified selectors through `ci_drift_heal` discovery instead of collapsing them back to ambiguous leaves
- fail closed with actionable diagnostics for genuinely ambiguous or unmapped managed candidates

Closes #2059.

## Root cause

#1677 / #1683 made the sync resolver canonical-path aware, but the auto-heal changed-file boundary still inferred test ownership from filenames alone. Root tests emitted lossy aliases (`cli`, `commands_generate`, `core_dump`, `core_errors`), and `detect_drift` also dropped the disambiguating `pdd/cli` selector while matching discovered prompts. #2013 fixed only reverse-include paths. Open #1971 changes sibling sync/architecture semantics; this PR deliberately avoids those files.

## TDD evidence

Red commits were pushed before the fix:

- `80eea77d3` — detector regression: **6 failed, 2 passed** on the buggy baseline
- `9fedf6fd6` — detector→drift-heal boundary: **1 failed**, showing `pdd/cli` was dropped

Green validation:

- `pytest -q tests/test_ci_detect_changed_modules.py tests/test_ci_drift_heal.py` — **234 passed**
- `pytest -q tests/test_ci_drift_heal.py::TestParseArgs tests/test_ci_drift_heal_e2e.py` — **11 passed**
- `python -m compileall -q scripts/ci_detect_changed_modules.py pdd/ci_detect_changed_modules.py pdd/ci_drift_heal.py` — passed
- `git diff --check` — passed
- focused pylint for the detector and changed drift-heal symbols — **10.00/10**

## Risk

The pdd_cloud auto-heal job uses a trusted private copy of the detector. This public fix must be mirrored/deployed there before hosted auto-heal receives the new normalization; this PR itself performs no deployment or cloud mutation.

### CODE REVIEW

**Comment by gltanaka** [comment]
## Codex adversarial review — PR #2060 @ ef861e1ffa35e111031b80b40b7ffe8f95f5e025

**Verdict:** DON'T MERGE
**Reviewed base:** `origin/main` @ `d500253f675cd08e61aa56817724dd048dc5e652`
**Findings:** 2 total (1 P1, 1 P2)

### P1 (blocking)
- [P1] The detector still rejects the actual #2058 change set it is meant to unblock — `scripts/ci_detect_changed_modules.py:285-317`, mirrored in `pdd/ci_detect_changed_modules.py`; reduced fixture at `tests/test_ci_detect_changed_modules.py:479-510`
  Replaying all 14 files from PR #2058 with reverse-dependency discovery stubbed produces `ValueError: unmapped auto-heal module candidate for tests/commands/test_evidence.py: 'commands/evidence'`. The same resolver independently rejects `tests/test_generate_estimate_accuracy.py`, `tests/test_grounding_generate_evidence.py`, `tests/test_grounding_test_plan.py`, and `tests/test_issue_826_snapshot_touchpoint.py`. These are legitimate cross-cutting/issue regression tests, but the implementation only resolves path-derived aliases; it does not implement the explicit-import ownership mapping promised in the issue investigation. The new regression fixture omits every one of these #2058 files, so hosted auto-heal remains blocked before drift healing.
  **Suggested fix:** Add the complete #2058 file list as the load-bearing regression and resolve non-conventional tests from exact local imports/canonical ownership (supporting multiple owners where necessary), reserving the unmapped failure for files with genuinely no canonical ownership evidence.

### P2 (non-blocking)
- [P2] Canonical selector matching performs a full path resolution once per discovered prompt — `pdd/ci_drift_heal.py:459-493`
  For every requested selector, `_select_requested_modules` calls `_resolve_paths(requested, language)` inside the loop over all discovered units even though the result depends only on `(requested, language)`. This repository discovers 309 units but only 16 languages. A real two-selector probe generated thousands of resolver log lines and had not completed after more than 60 seconds; #2058's multi-module candidate set multiplies that work further on the required auto-heal path.
  **Suggested fix:** Resolve/cache once per unique `(requested, language)`, then compare the cached canonical prompt against discovered paths; add a bounded call-count regression.

### Evidence
- `pytest -q tests/test_ci_detect_changed_modules.py tests/test_ci_drift_heal.py`: **235 passed**.
- Exact detector CLI for this PR: `ci_drift_heal`, exit 0.
- Exact #2058 changed-file replay: deterministic unmapped failure at `tests/commands/test_evidence.py`.
- Independent probes confirm four additional #2058 regression tests are unmapped.
- Source/package detector diff is header-only; behavior is otherwise byte-equivalent.
- `git diff --check origin/main...HEAD`: pass.
- Prior scope reviewed: #1677 establishes path-qualified identity/fail-closed ambiguity; #2013 fixes reverse includes; open #1971 owns the broader sync resolver. This finding stays at #2060's auto-heal boundary.

**Remaining risk:** Hosted auto-heal and unit checks were still pending at review time. No provider or deployment path was exercised.

**Comment by gltanaka** [comment]
Addressed the independent P1/P2 review at `11fdcbb3d` (red tests were pushed first in `c8d9f8b81`).

### P1 — complete #2058 replay

The regression now replays all 15 paths reported by GitHub for #2058, including the five previously unmapped cross-cutting tests. Non-conventional tests fall back only to exact canonical ownership from parsed `pdd...` imports; multiple imported owners are retained. Filename ambiguity still fails before import fallback, and a test with no canonical import evidence remains unmapped/fail-closed.

Exact replay output:

`commands/analysis,commands/checkup,commands/generate,commands/maintenance,commands/misc,commands/modify,commands/replay,commands/utility,context_snapshot,core/cli,core/dump,core/errors,evidence_manifest,grounding_policy,llm_invoke,pdd/cli,preprocess`

The trusted-mirror sibling finding is also covered: excluded reverse candidates such as `src/clients/github_client` are filtered before ownership resolution. Direct exclusions and script/package parity remain covered.

### P2 — bounded selector resolution

`_select_requested_modules` now caches by `(requested, language)`. The deterministic regression uses 120 discovered units across two languages and two selectors: resolver calls fell from **240 red** to **4 green**. A real-repository probe over **309 discovered units** selected `pdd/cli` and `core/cli` in **2.083s**, comfortably below 60s.

### Evidence

- Red correction suite on `c8d9f8b81`: **5 failed** (both detector copies, reverse exclusion, 240-vs-4 call bound).
- `pytest -q tests/test_ci_detect_changed_modules.py tests/test_ci_drift_heal.py`: **240 passed**.
- CLI parse/e2e boundary: **11 passed**.
- Exact #2058 replay: exit 0 with the canonical list above.
- Script/package detector parity: header-only.
- Compile, `git diff --check`, focused detector lint, and changed drift-heal static checks: passed; pylint **10.00/10**.

No merge or deployment was performed.

File: scripts/ci_detect_changed_modules.py [hunk 1]
```diff
@@ -180,6 +191,132 @@ def _basename_from_path(path: str) -> str | None:
...
+    )
+
+
+def _module_from_path(path: str, ownership: _ModuleOwnership) -> str | None:
+    """Map one changed path to its architecture-owned canonical module."""
+    normalized = _normalize_repo_path(path)
+    if normalized in ownership.code_paths:
+        return ownership.code_paths[normalized]
+    if normalized in ownership.prompt_paths:
+        return ownership.prompt_paths[normalized]
```
[the remark this answers is not in the document]
Comment (gltanaka) [reply scripts/ci_detect_changed_modules.py]: Addressed in `ef861e1ff` for exact direct architecture/prompt paths, with `test_detect_keeps_architecture_owned_exclusions_out` guarding the regression. The correction cycle also covers the sibling reverse-dependency path in `11fdcbb3d`: excluded candidates are filtered before ownership resolution, including `src/clients/github_client`.

**Review by gltanaka — COMMENTED** [review]
## Independent critical re-review — PR #2060 @ `11fdcbb3d22e2294a5f4febc3398aa45beb634ae`

**Verdict: DON'T MERGE**  
**Base:** `d500253f675cd08e61aa56817724dd048dc5e652`  
**Findings:** 1 (P2)

### P2 — exact head no longer passes the required diff hygiene gate

The hosted auto-heal checkpoint commit `47f7ca16b` added ten trailing-whitespace errors to `context/ci_drift_heal_example.py` (lines 2, 38, 43, 44, 54, 60, 65, 88, 90, and 98 in the changed file). Consequently:

```text
git diff --check d500253f675cd08e61aa56817724dd048dc5e652...11fdcbb3d22e2294a5f4febc3398aa45beb634ae
```

exits non-zero. This invalidates the earlier exact-head evidence claiming `git diff --check` passed and leaves generated release content with avoidable whitespace churn. Strip the trailing whitespace and rerun the exact-head gate. If a clean regeneration reproducibly reintroduces it, track the generator behavior separately rather than accepting a dirty release diff.

### Functional verification

- `pytest -q tests/test_ci_detect_changed_modules.py tests/test_ci_drift_heal.py`: **240 passed**.
- Complete #2058 changed-file replay: canonical 17-owner output; the prior P1 is fixed.
- Resolver caching regression: bounded by selector × unique language; the prior P2 is fixed.
- Direct AST probes on both detector copies: aliased imports, `from pdd... import ...`, multiple owners, relative imports, non-PDD imports, ambiguous aliases, and unmapped tests all behaved safely/fail-closed.
- `src/clients/github_client` reverse exclusion: covered and passing.
- Script/package parity: exact except the documented packaged header.
- `compileall`: passed.
- Prompt hardlink/parity: exact.
- Scope review: no changes to `sync_determine_operation` resolver ownership owned by #1971.

**Remaining risk:** hosted checks were still pending during review. No merge, deployment, provider call, or production mutation was performed.

**Review by gltanaka — COMMENTED** [review]
## Independent correction re-review — PR #2060 @ `52490662101d2cd7a455419e56a7e9e76f7e7482`

**Verdict: MERGE / READY**  
**Base:** `d500253f675cd08e61aa56817724dd048dc5e652`  
**Findings:** none

The delta from previously reviewed head `11fdcbb3d22e2294a5f4febc3398aa45beb634ae` is limited to `context/ci_drift_heal_example.py`. `git diff --ignore-space-at-eol --exit-code 11fdcbb3d...52490662` returns 0, confirming there is no functional/content delta beyond end-of-line whitespace normalization.

Verification:
- `git diff --check origin/main...HEAD`: **pass** (exact base `d500253f`).
- `pytest -q tests/test_ci_detect_changed_modules.py tests/test_ci_drift_heal.py`: **240 passed**.
- Changed files since prior review: **1**, generated example only.
- Functional detector/drift-heal sources, tests, prompt, metadata, and script/package mirror: unchanged from the prior reviewed head.
- Worktree clean; no merge, deployment, provider call, or production mutation performed.

Hosted exact-head checks were still running at review time and remain the merge gate.

**Review by gltanaka — COMMENTED** [review]
## Independent hosted-auto-heal review — PR #2060 @ `28ae89dd1d61e5c6a263991ef09806b6e2b630bd`

**Verdict: DON'T MERGE**  
**Base:** `d500253f675cd08e61aa56817724dd048dc5e652`  
**Findings:** 2 (1 P1, 1 P2)

### P1 — the regenerated example's primary demo fails, masks the failure, and mutates the checkout

`setup_mock_environment()` writes `output/.pddrc` and `output/prompts/calculator_python.prompt`, but `demonstrate_dry_run()` never changes into that subproject or points discovery at it. It calls `main(modules=["calculator"], dry_run=True)` from the repository root, where `calculator` is not a discovered synchronization unit.

An isolated-clone execution produced:

```text
dry-run: metadata_stale=0 conflicts=0 unbaselined=0 failures=0
Dry-run execution exited with code: 1
...
Example run completed successfully.
```

The process exits **0** because the nonzero dry-run result is only printed and then discarded. The apparently clean text summary also omits the underlying report failure. Execution added ignored `output/` (including the two mock files), `litellm_cache.sqlite/`, and Python cache artifacts to the checkout, contradicting the example's stated no-write dry-run behavior and the requested safe execution contract.

Make the fixture participate in the actual discovery context (or remove the disconnected fixture), propagate a nonzero `main()` result, and do not print overall success after failure. Run writable demonstrations in a temporary directory rather than the repository checkout.

### P2 — generated-file formatting is incomplete

`context/ci_drift_heal_example.py` has no final newline, and `os` is imported but unused (`pylint W0611`). Add the newline and remove the dead import so the generated example meets normal source hygiene. (`git diff --check` passes because it does not reject missing EOF newlines.)

### Verified clear

- Provider-free isolated execution: no LLM/provider request observed.
- `commit_and_push([])`: API is accurate and returns before staging, commit, or push.
- Imports and signatures: valid; example compiles and executes.
- Metadata: prompt/code/example/test/test-files/include-dependency hashes all exactly match the current tree.
- Detector implementation, prompt, mirror, and detector tests are byte-identical to approved `52490662`.
- `pytest -q tests/test_ci_detect_changed_modules.py tests/test_ci_drift_heal.py`: **240 passed**.
- `git diff --check origin/main...HEAD`: **pass**.
- Worktree clean; no merge, deployment, provider call, or production mutation performed.

Hosted checks for this new exact head were still running at review time.

**Comment by gltanaka** [comment]
## TDD red — protected human-owned infrastructure @ `00d128f3b`

Hosted #2065 / Cloud Build `d7188724-68ac-43de-902f-c7199341290c` exposed that the staged fail-closed detector treats protected `pdd/sync_core/verification.py` as an unmapped generated candidate.

Focused trusted-script + packaged-copy result:

```text
2 failed, 2 passed
```

Both exact protected-base `verification.py` cases fail with `unmapped auto-heal module candidate`; both candidate-only self-authorization controls already pass (the unknown path remains rejected). This commit contains tests only.

**Comment by gltanaka** [comment]
## General detector correction — `15a131c81cf2b6cfd6ac1cfc2cbb502b748804f6`

The detector now uses architecture/prompt ownership first. Only when a path is otherwise unmapped does it read `.pdd/sync-ownership.json` from the exact left-hand protected Git ref and exempt a strict, exact `HUMAN_OWNED`/`pdd-maintainers`/human-maintained row. Candidate policy bytes, wildcards, malformed rows, and genuinely unknown PDD sources cannot authorize the candidate.

Evidence:

- red `00d128f3b`: 2 failed / 2 passed
- green exact + negative controls: 8 passed
- exact #2065 hosted-equivalent detector invocation: exit 0, empty module set
- trusted script/package behavior parity: pass
- scoped pylint (excluding four pre-existing file diagnostics plus intentional mirror duplication): 10.00/10
- `git diff --check`: pass

The full detector file in one process remains resource-sensitive in this worktree (the prior hosted unit run was already failing before this correction); hosted exact-head checks are running. No merge, pdd_cloud mutation, context restage, provider call, or deploy performed.

**Comment by gltanaka** [comment]
## Clean stacked reconstruction — `4ca23a57a`

Base: PR #2063 `09a11986c` (itself on #2065 `dffaa346e`). Retained logical history from `80eea77d3`, `9fedf6fd6`, `2fb7db608`, `ef861e1ff`, `c8d9f8b81`, `11fdcbb3d`, plus human-ownership TDD/fix `00d128f3b`/`15a131c81`. Dropped checkpoint churn `47f7ca16b`, `524906621`, `28ae89dd1`. Added exact protected detector-profile consumption `29230aa26` and safe metadata refresh `4ca23a57a`.

No cherry-pick or rebase conflicts. The seven logical implementation/test files match the reviewed old #2060 bytes. #2063 prompt/example/contract and run-report bytes remain exact. Metadata changes only the legitimate code/test hashes, preserves prompt/example hashes, and retains all three test files.

Evidence: detector 41/41; drift 203/203; contract 17/17; protected-human + negative controls 8/8; hostile twice/no-mutation 8/8; metadata hashes/test_files exact; script/package mirror exact; real detector CLI outputs `ci_drift_heal`; real read-only sync returns `all_synced`; exact base-to-head 466 profiles, coverage 1.0, no manifest/profile errors; detector source pylint 10.00/10; compile/diff/clean-tree pass.

No merge or deployment performed.

**Comment by gltanaka** [comment]
## Independent critical review — reconstructed stack

**Verdict: MERGE / stack-ready**, at exact base `09a11986c634df4a33907b07f067e09c504f9ee3` and head `4ca23a57a2f3bc5fc4b0bb0cde85325f6fede14f`. Full normal CI remains required after the planned squash/rebase onto `main`.

### Findings

No blocking or non-blocking code findings.

- Each of the eight reconstructed logical commits has the same stable patch ID as the previously reviewed `80eea77d3…15a131c81` sequence. The unsafe generated checkpoints `47f7ca16b`, `524906621`, and `28ae89dd1` are absent. The seven logical implementation/test files are byte-identical to reviewed `15a131c81`.
- #2063 prompt, safe example, contract test, and `ci_drift_heal_python_run.json` bytes are unchanged. Drift metadata changes only the legitimate detector code/test hashes, preserves prompt/example/include provenance, and retains all three test files with exact hashes.
- Detector/profile rotation consumption matches the single protected `ci_detect_changed_modules` rule and all four exact base/head policy/prompt SHA bindings. Exact `09a11986… → 4ca23a57…` evaluation is 466/466 profiles, zero manifest/profile errors, zero unaccounted paths, coverage 1.0. Candidate-only transition authority and wrong-binding/validator-remap controls reject.
- Canonical generated paths resolve; ambiguous and genuinely unmapped paths fail closed; candidate-only human ownership cannot authorize; exact protected human-owned `verification.py` is skipped. Exact #2065 detector replay exits 0 with an empty module set.
- Script/package detector parity is exact except the documented three-line package header. Exact stack detector output is `ci_drift_heal`.

### Tests and safety evidence

- Detector: **41 passed**.
- Drift-heal: **203 passed**.
- Safe example contract: **17 passed**.
- Focused human-owned/unmapped/ambiguous/candidate-only boundary: **8 passed**.
- Hostile AST/Git-shim plus twice/no-mutation subset: **10 passed**.
- Requirement self-authorization/wrong-binding/validator-remap controls: **3 passed**.
- Real read-only `ci_drift_heal` sync with tests skipped: `all_synced`; worktree remains clean.
- Compile and exact diff checks pass. Detector pylint is 9.92/10 solely from four pre-existing diagnostics; no new diagnostic was introduced.

### Residual risk / hosted state

The current `heal`/`auto-heal` checks are green but Cloud Build `9d6580ee-5392-4982-a58d-1b9cd24d8e59` explicitly skipped execution because this stacked PR targets `agent/2061-safe-drift-example`, not `main`. The intentionally preserved #2063 run report predates the new drift tests, so normal test-aware sync currently requests `test_extend`; this makes the post-squash/rebase full unit and hosted auto-heal runs the required final merge gate.

No edit, merge, provider call, or deployment was performed.

**Comment by gltanaka** [comment]
## Main-targeted stack integration — `79eb16bbf`

Rebased only the reviewed #2060 delta from old base `09a11986c` / head `4ca23a57a` onto squash-merged `main` `23ff58366`.

### Topology and safety

- `git range-diff`: all ten reconstructed logical commits map one-to-one with identical patch IDs.
- Retained the canonical candidate, exclusion, complete #2058 ownership, human-owned detector, profile-consumption, and metadata-refresh changes.
- Unsafe auto-heal checkpoints `47f7ca16b`, `524906621`, and `28ae89dd1` remain absent.
- Rebase conflicts: none.
- #2060 source delta remains the intended nine files. One additional main-integration test-only commit updates the new #2065 rollout assertion from dormant base bytes to the consumed, preauthorized head bytes.

### Test evidence

- Detector: **41 passed**
- Drift heal: **203 passed**
- Safe example contract: **17 passed**
- Combined exact suite: **261 passed**
- Protected rollout policy: red before the integration assertion correction (old base prompt expected), then **8 passed**; current denominator remains **466**, coverage **1.0**, with zero invalid/unaccounted paths.
- Canonical/unmapped/ambiguous/candidate-only/protected-human and complete #2058 replay controls are included in the green detector suite.
- JSON parsing, compileall, script/package parity, and `git diff --check origin/main...HEAD`: pass.
- Detector pylint: **9.92/10**, only the same four previously documented diagnostics; no new diagnostic.

The preserved stale `ci_drift_heal` run report was not regenerated or manufactured. A local full Vitest-file run is non-authoritative because this workstation's `pdd` environment lacks the exact `tree-sitter*` packages (`Vitest JavaScript parser is unavailable`); exact-head hosted Unit CI is the required #2071 integration signal.

Force-push used an explicit lease against old remote head `4ca23a57a`. PR base is now `main`; local and remote heads both equal `79eb16bbf01b04e07fe76e165a5be5d0f8f386df`. No merge or deployment was performed.

**Comment by gltanaka** [comment]
## Independent critical review — PR #2060 @ `94eebf6dddf56af3276417c2338466539b4723a6`

**Verdict: DO NOT MERGE**
**Findings: 2 total (2 P1, 0 P2)**

The assigned head `79eb16bbf01b04e07fe76e165a5be5d0f8f386df` passed the cumulative review, but the branch changed during review. This verdict applies to the new exact head above.

### P1 (blocking)

- [P1] Auto-heal reintroduces the unsafe generated example that this reconstruction explicitly removed — `context/ci_drift_heal_example.py:29-139`
  Commit `94eebf6dd` replaces the reviewed provider-free, bounded child-process example with an in-process call after `os.environ.copy()`; the constructed `custom_env` is never applied. It removes `invoke_main`, `run_dry_run`, `create_disposable_project`, the 30-second timeout, credential/config stripping, and caller/source Git-state guards required by `tests/test_ci_drift_heal_example_contract.py`. The checked-in contract directly calls those removed APIs and therefore cannot pass at this head. This is the same semantic checkpoint churn intentionally excluded from `47f7ca16b`, `524906621`, and `28ae89dd1`, now reintroduced by the bot.
  **Suggested fix:** Remove/revert only `94eebf6dd`, restore exact reviewed head `79eb16bbf`, and prevent auto-heal from regenerating this protected safe example.

- [P1] Generated metadata drops reviewed test ownership and trusted run evidence — `.pdd/meta/ci_drift_heal_python.json:8-10`, `.pdd/meta/ci_drift_heal_python_run.json`
  The bot shrinks `test_files` from the three reviewed files to only `test_ci_drift_heal.py` and deletes the run report. That loses ownership/evidence for `test_ci_drift_heal_e2e.py` and `test_ci_drift_heal_example_contract.py`, recreating the stale `test_extend`/regeneration loop the reconstructed stack was designed to expose without accepting its generated rewrite.
  **Suggested fix:** Restore the exact three-file metadata and preserved run-report bytes from `79eb16bbf`; do not accept bot-generated metadata churn.

### Verified before the head changed

- Old ten-commit stack range-diffs 1:1 with identical patches; unsafe checkpoint commits are not ancestors.
- Exact `origin/main → 79eb16bbf` protected transition: 466 profiles, coverage 1.0, zero manifest/profile errors and zero unaccounted paths.
- Detector/drift/example-contract/rollout suite at `79eb16bbf`: 268 passed; one unrelated final clone-based rollout test failed only because the filesystem had 1.4 GiB free and Git returned `ENOSPC`.
- JSON/compile/mirror/hash/diff checks passed at `79eb16bbf`.
- Hosted checks must be evaluated again only at a corrected exact head.

**Comment by gltanaka** [comment]
## Correction for hosted auto-heal overwrite — `5b5c6ad60`

Cloud Build `dbc7d806-a530-4776-ae01-a487c86c609e` confirmed the root cause: valid code drift selected `ci_drift_heal`, then `_heal_update` unconditionally ran a follow-up `pdd example` even though the exact protected-base ownership policy marks `context/ci_drift_heal_example.py` human-owned. That follow-up regenerated the unsafe example, narrowed fingerprint test ownership, and cleared the protected run report.

### TDD and correction

- Red commit `dbc6740af`: protected example/evidence regression failed on bot-head semantics because the example command ran and architecture evidence narrowed.
- Green commit `5b5c6ad60`: strict exact protected-base ownership only (no candidate authority, wildcard, traversal, or malformed rows); protected example bytes are restored from the left-hand Git ref; protected fingerprint test ownership is retained with current hashes; protected run evidence is restored; follow-up example generation is skipped.
- Restored the reviewed hardened example and run report byte-for-byte from `79eb16bbf`.
- Restored three-file metadata ownership and refreshed only legitimate current code/test hashes.

### Evidence

- Detector + drift-heal + safe-example contract: **262 passed** (**41 / 204 / 17**).
- Protected rollout: **8 passed** before the final fail-closed parsing hardening. After that non-manifest change, the six non-clone tests passed again; the two clone fixtures failed only at `git clone` with `ENOSPC`. The 466-profile/full-coverage/zero-invalid/unaccounted assertions are among the passing current tests.
- Exact regression: protected example, multi-test fingerprint mapping, and run report all survive simulated destructive metadata/example behavior.
- Metadata code/example/test hashes and all three declared test files: exact.
- JSON, compile, and `git diff --check`: pass.
- Final diff from pre-bot reviewed head `79eb16bbf` is limited to `.pdd/meta/ci_drift_heal_python.json`, `pdd/ci_drift_heal.py`, and `tests/test_ci_drift_heal.py`; hardened example/run bytes are unchanged.

Explicit-lease push advanced remote from `94eebf6dd` to `5b5c6ad606fdc7acee4a015d7cf62172027d97c5`. No merge or service deployment performed. Hosted exact-head Unit/auto-heal and fresh independent review remain required.
